src/richie/apps/courses/api.py

```diff
@@ -159,18 +159,19 @@ def sync_course_run(data):
 
     # Look for the course targeted by the resource link
     course_code = normalize_code(lms.extract_course_code(data))
-    try:
-        course = Course.objects.get(
+    course = (
+        Course.objects.filter(
             code=course_code,
             extended_object__publisher_is_draft=True,
             # Exclude snapshots
             extended_object__node__parent__cms_pages__course__isnull=True,
         )
-    except Course.DoesNotExist as exc:
+        # distinct fixes MySQL returning same course twice
+        .distinct().first()
+    )
+    if not course:
         # Create the course page in draft
-        raise ValidationError(
-            {"resource_link": [f"Unknown course: {course_code:s}."]}
-        ) from exc
+        raise ValidationError({"resource_link": [f"Unknown course: {course_code:s}."]})
 
     # Instantiate a new draft course run
     draft_course_run = CourseRun(
```

Comment:
I think it can be reproduced in a test

Reply:
@igobranco I push a commit where there is a test which reveals the issue.
In fact, the problem occurred once the page has been published... 😳 (The `extended_object__node__parent__cms_pages__course__isnull=True` targets both draft and published pages)

Reply:
Your test didn't has impact on the problem.
Nevertheless, I've still included in the code.